### PR TITLE
feat(sidebar): Mobile default improvements

### PR DESCRIPTION
**Pull request by gadenbuie**
The biggest change in this PR is to avoid a CSS animation for sidebars with `open="desktop"` when initializing the sidebar on mobile screen sizes. With this default, `layout_sidebar()` provides initially correct markup for open sidebars, but we close the sidebar when initialized. Previously, we would use the exact same code path as if the user had clicked the collapse toggle, but this caused a distracting animation on page load.

We now instead skip the transition – mostly use the same code path – but without adding the `.transitioning` class. In this way, the sidebar closes immediately but we don't lose the aria labels and other appropriate accessibility markup that we require. Here's a sketch of those changes:

1. When `open="desktop"`, on small screens, we now add the collapsed class, without including the `.transitioning` class and immediately invoke `_finalizeState()`.
2. All CSS animations in the sidebar now only apply when the `.transitioning` class is added before the state-changing class is toggled. This is achieved by enabling `--_transition-duration` only when the `.transitioning` class is added.
3. I did break the contract that we emit "initially correct markup for open sidebars" by adding `hidden` to the `open="desktop"` sidebar element. We remove this attribute if the sidebar should be open. This prevents some layout jumps from happening if the sidebar's contents are briefly shown before being closed.

While in this area, I also:

* Added `--bslib-sidedbar-mobile-max-width`, `--bslib-sidebar-mobile-opacity` and `--bslib-sidebar-mobile-box-shadow` CSS custom properties that be set globally (or `--_sidebar-mobile-*` for locally scoped changes). We don't use these yet, but they open the door for working with design variations of the sidebar on mobile.

* I adjusted the sidebar transition duration default from `500ms` to `300ms` for a snappier transition.

### CODE REVIEW

File: srcts/src/components/sidebar.ts [hunk 1]
```diff
@@ -314,7 +314,10 @@ class Sidebar {
       .getPropertyValue("--bslib-sidebar-js-init-collapsed");
 
     if (initCollapsed.trim() === "true") {
-      this.toggle("close");
+      container.classList.add(Sidebar.classes.COLLAPSE);
+      this._finalizeState();
+    } else {
+      sidebar.hidden = false;
     }
   }
```
Comment (cpsievert) [inline srcts/src/components/sidebar.ts:322]:
```suggestion
      container.classList.add(Sidebar.classes.COLLAPSE);
    }
    this._finalizeState();
```

Comment (cpsievert) [reply srcts/src/components/sidebar.ts:322]: More generally, this is reaching into the internal logic of `toggle()` and it'd be nice to have this become something like `toggle("close", immediate = true)`

File: inst/components/scss/sidebar.scss [hunk 2]
```diff
@@ -318,6 +325,19 @@ $bslib-sidebar-column-sidebar: Min(calc(100% - var(--_icon-size)), var(--_sideba
         left: unset;
       }
 
+      .sidebar {
+        opacity: var(--_sidebar-mobile-opacity);
+        max-width: var(--_sidebar-mobile-max-width);
+        box-shadow: var(--_sidebar-mobile-box-shadow);
+      }
+
```
Comment (cpsievert) [inline inst/components/scss/sidebar.scss:333]: Let's add a comment about this being connected to `max-width`

File: inst/components/scss/sidebar.scss [hunk 3]
```diff
@@ -318,6 +325,19 @@ $bslib-sidebar-column-sidebar: Min(calc(100% - var(--_icon-size)), var(--_sideba
         left: unset;
       }
 
+      .sidebar {
+        opacity: var(--_sidebar-mobile-opacity);
+        max-width: var(--_sidebar-mobile-max-width);
+        box-shadow: var(--_sidebar-mobile-box-shadow);
+      }
```
Comment (cpsievert) [inline inst/components/scss/sidebar.scss:332]: If these are inherited properties, do we need to default the CSS variable to `inherit`?

File: inst/components/scss/sidebar.scss [hunk 4]
```diff
@@ -6,7 +6,7 @@ $bslib-sidebar-column-sidebar: Min(calc(100% - var(--_icon-size)), var(--_sideba
 
 
 .bslib-sidebar-layout {
-  --_transition-duration: max(var(--bslib-sidebar-transition-duration, 500ms), 5ms);
+  --_transition-duration: ; // Set below for .transitioning only
```
Comment (cpsievert) [inline inst/components/scss/sidebar.scss:9]:
```suggestion
  --_transition-duration: 0ms; // Below in .transitioning, we bring the actual duration 
```

File: inst/components/scss/sidebar.scss [hunk 5]
```diff
@@ -335,7 +355,7 @@ $bslib-sidebar-column-sidebar: Min(calc(100% - var(--_icon-size)), var(--_sideba
 
       // Make main contents transparent while sidebar is expanded
       > .main {
-        opacity: 0;
+        // opacity: 0;
```
Comment (cpsievert) [inline inst/components/scss/sidebar.scss:358]:
```suggestion
```

File: inst/components/scss/sidebar.scss (hunk 3 above)
Comment (gadenbuie) [reply inst/components/scss/sidebar.scss:332]: Good news: none of these are inherited properties. I'll added defaults, but I think we're okay here.

File: srcts/src/components/sidebar.ts (hunk 1 above)
Comment (gadenbuie) [reply srcts/src/components/sidebar.ts]: Done in https://github.com/rstudio/bslib/pull/907/commits/79de52212152619753bdaf7a4d5224dd80efe35d, I moved all logic into the `.toggle()` method, which is now called with the appropriate `method`, i.e. for both `"open"` and `"close"`.

File: inst/components/scss/sidebar.scss [hunk 6]
```diff
@@ -318,6 +325,20 @@ $bslib-sidebar-column-sidebar: Min(calc(100% - var(--_icon-size)), var(--_sideba
         left: unset;
       }
 
+      .sidebar {
```
Comment (cpsievert) [inline inst/components/scss/sidebar.scss:328]: Not sure if this is important or not, but we seem to do this consistently elsewhere?

```suggestion
      > .sidebar {
```

Comment (gadenbuie) [reply inst/components/scss/sidebar.scss:328]: Good catch, yes we need the `>`

**Pull request merged**